gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinBaseOrcWriter.java

```diff
@@ -183,9 +241,52 @@ public void commit()
       throws IOException {
     closeInternal();
     super.commit();
+    properties.setProp(ORC_WRITER_ESTIMATED_RECORD_SIZE, String.valueOf(estimatedRecordSize));
+    properties.setProp(ORC_WRITER_ESTIMATED_BYTES_ALLOCATED_CONVERTER_MEMORY, String.valueOf(this.converterMemoryManager.getConverterBufferTotalSize()));
   }
 
   /**
+   * Modifies the size of the writer buffer based on the average size of the records written so far.
+   * If the new batch size is smaller than size of the current batch, the current batch is flushed.
```

Comment:
Is this comment correct? The current code seems to:
1. Calculate the new batch size
2. If the new batch size is 20% greater or smaller, then we flush
3. and then ensure the new row batch is of the new batchsize

We seem to flush for larger and smaller

Reply:
Yeah I used to have it only for smaller but realized it's easier to manage certain edge cases if it just flushes every time it resizes the batch, will update the comment